hcl2template/types.required_plugins.go

```diff
@@ -0,0 +1,224 @@
+package hcl2template
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/packer/hcl2template/addrs"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func (cfg *PackerConfig) decodeRequiredPluginsBlock(f *hcl.File) hcl.Diagnostics {
+	var diags hcl.Diagnostics
+
+	content, moreDiags := f.Body.Content(configSchema)
+	diags = append(diags, moreDiags...)
+
+	for _, block := range content.Blocks {
+		switch block.Type {
+		case packerLabel:
+			content, contentDiags := block.Body.Content(packerBlockSchema)
+			diags = append(diags, contentDiags...)
+
+			// We ignore "packer_version"" here because
+			// sniffCoreVersionRequirements already dealt with that
+
+			for _, innerBlock := range content.Blocks {
+				switch innerBlock.Type {
+				case "required_plugins":
+					reqs, reqsDiags := decodeRequiredPluginsBlock(innerBlock)
+					diags = append(diags, reqsDiags...)
+					cfg.Packer.RequiredPlugins = append(cfg.Packer.RequiredPlugins, reqs)
+				default:
+					continue
+				}
+
+			}
+		}
+	}
+	return diags
+}
+
+func (cfg *PackerConfig) decodeImplicitRequiredPluginsBlocks(f *hcl.File) hcl.Diagnostics {
+	// when a plugin is used but not defined in the required plugin blocks, it
+	// is 'implicitly used'. Here we read common configuration blocks to try to
+	// guess plugins.
```

Comment:
so is this how the json configs will work?

Reply:
Yes, I think so !

Reply:
Though we'd have to implement that part JSON side for it to work.

Reply:
My thinking here would be to release packer with `json + plugin-getter + plugins`, then release packer with `plugin-getter` ( so deprecate json and remove core plugins )